vignettes/qgisprocess.Rmd

```diff
@@ -0,0 +1,271 @@
+```
+
+We can find out about the arguments again with the help of `qgis_arguments()`.
+
+```{r args}
+qgis_arguments("grass7:r.slope.aspect")
+```
+
+`qgis_outputs()` shows the different outputs that will be calculated:
+
+```{r outputs}
+qgis_outputs("grass7:r.slope.aspect")
+```
+
+Now let us calculate the terrain attributes.
+
+```{r slopeaspect, message=FALSE}
+library("terra")
+# attach digital elevation model from Mt. Mongón (Peru)
+dem <- rast(system.file("raster/dem.tif", package = "spDataLarge"))
+
+info <- qgis_run_algorithm(alg = "grass7:r.slope.aspect", elevation = dem)
+                           
+```
+
+Just printing the `info` object shows which output objects have been made:
+
+```{r info}
+info
+```
+
+Combine these output rasters as a multi-layered `SpatRaster` object and plot it:
+
+```{r combine1, message=FALSE}
+# just keep the names of output rasters
+nms <- qgis_outputs("grass7:r.slope.aspect")$name
+# read in the output rasters 
+r <- info[nms] |>
+  unlist() |>
+  terra::rast() |>
+  as.numeric()
+names(r) <- nms
+# plot the output
+plot(r)
+```
+
+An alternative way to combine the rasters is given below.
+
+```{r combine2, message=FALSE}
+r <- lapply(info[nms], \(x) as.numeric(qgis_as_terra(x))) |> 
+  rast()
+plot(r)
+```
+
+## Piping
+
+`qgis_process` does not lend itself naturally to piping because its first argument is the name of a geoalgorithm instead of a data object.
+`qgis_pipe()` circumvents this by accepting a `.data` object as its first argument, and pipes this data object into the the first argument of a geoalgorithm assuming that the specified geoalgorithm in fact needs a data input object as its first argument.
+
+```{r buf-pipe}
+system.file("longlake/longlake_depth.gpkg", package = "qgisprocess") |>
+  qgis_pipe("native:buffer", DISTANCE = 50)
+```
+
+In case of a `qgis_result` object, `qgis_pipe()` automatically tries to select an element named `OUTPUT`.
+However, if the output has another name (e.g., `DEM_PREPROC` as in the example below) or if there are multiple output elements to choose from (e.g., `sagang:sagawetnessindex` has four output rasters, check with `qgis_outputs("sagang:sagawetnessindex")`, you can specify the wanted output object via the `.select` argument.
+
+```{r qgis-pipe}
+dem <- system.file("raster/dem.tif", package = "spDataLarge")
+# in case you need to enable the SAGA next generation algorithms, run the following lines:
+# qgis_plugins()
+# qgis_enable_plugins("processing_saga_nextgen")
+qgis_run_algorithm(algorithm = "sagang:sinkremoval", DEM = dem, 
+                   METHOD = 1) |>
+  qgis_pipe("sagang:sagawetnessindex", .select = "DEM_PREPROC")
+```
+
+When piping, `qgis_pipe()` automatically cleans up after you by deleting intermediate results.
+This avoids cluttering your system when running geoalgorithms on large spatial data files.
+To turn of this behavior, set `.clean` to `FALSE`.
+
+Of course, you can also use the native pipe.
```

Comment:
```suggestion
Of course, you can also pipe to `qgis_run_algorithm()` by manually extracting the `OUTPUT` object and redirecting it to the appropriate input argument of the next processing step.
```

The choice here is not about using (native) pipes or not, as pipes are used with `qgis_pipe()` too. Also 'native' vs **magrittr**'s pipe is not relevant for the point made, except for syntax difference if using the placeholder.